### ISSUE TITLE
Document error returns

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#1780]
Running `cargo clippy -- -Dclippy::missing_errors_doc` gives a bunch of warnings. We should consider documenting all error returns.

**Comment by startup-dreamer** [comment]
@tcharding I would like to work on this issue can you please assign me

**Comment by tcharding** [comment]
Go for it man, we don't use the assign button.

**Comment by Kixunil** [comment]
I would like to enable this lint in CI after fixing it.

**Comment by tcharding** [comment]
I agree, as long as we don't get folks adding docs that add zero benefit just to pass the lint.

**Comment by startup-dreamer** [comment]
Understood i will make comprehensible changes and make a PR if the changes doesn't satisfy the norms than it might help some other contributor :)

**Comment by tcharding** [comment]
This is included in #3825

**Comment by tcharding** [comment]
Woops #3825 is only for `units`. This issue is needed for each crate before it hits 1.0
